### ISSUE TITLE
Reproducing dev set result with roberta-large

### ISSUE THREAD

**Issue by jaehunjung1** [PlusLabNLP/Com2Sense#1]
Hi, I am trying to reproduce the dev set accuracy in Table 11 of the paper with Roberta-large (**62.81**). 

I have run several times with the demo script in README.md, but this yields an accuracy of **51.04** which is significantly lower than expected. I have also tried with the hyperparameters stated in the appendix of the paper, but the accuracy seems to be around 51~52 as well. 

Can you provide a specific instruction on how to reproduce the result in the paper?

**Comment by Axe--** [comment]
Hey, thanks for reaching out. Will update you by tomorrow.

**Comment by Palipoor** [comment]
Hi @sharkmir1 
I think we should update the scripts in the readme to reflect the hyperparameters from the paper, but I was able to reproduce the results using this script, which is the same but with batch size 32 and sequence length 80 from the paper:

`python3 main.py --mode train --dataset com2sense --model roberta-large --expt_dir ./results --expt_name roberta --run_name demo --seq_len 80 --epochs 100 --batch_size 32 --acc_step 4 --lr 1e-5 --log_interval 500 --gpu_ids 0,1,2,3 --use_amp T -data_parallel`

How many epochs did you let the training continue for? Can you show your script or training logs?

**Comment by Palipoor** [comment]
This is my train_log.txt file:

```
python3 main.py
--mode train
--expt_dir ./results
--expt_name roberta
--run_name demo
--test_file test
--model roberta-large
--num_layers -1
--seq_len 80
--num_cls 2
--ckpt None
--pred_file results.csv
--dataset com2sense
--lr 1e-05
--epochs 100
--batch_size 32
--acc_step 4
--log_interval 500
--save_interval 30000
--val_size 2048
--gpu_ids 0,1,2,3
--data_parallel True
--use_amp True
--cpu False
--num_workers 1


Train: 1608 
Validation: 782

Validation Accuracy is computed using 782 samples. See --val_size
No. of Classes: 2

-------------------------------------------------------------------------

After 1 epoch:
Validation Accuracy: 49.87 %  || Validation Loss: 0.7271

** Best Performing Model: 49.87 ** 
Saving weights at ./results/roberta/demo/ep_1_stp_51_acc_49.8698_roberta_large.pth
-------------------------------------------------------------------------


-------------------------------------------------------------------------

After 2 epoch:
Validation Accuracy: 50.78 %  || Validation Loss: 0.6995

** Best Performing Model: 50.78 ** 
Saving weights at ./results/roberta/demo/ep_2_stp_101_acc_50.7812_roberta_large.pth
-------------------------------------------------------------------------


-------------------------------------------------------------------------

After 3 epoch:
Validation Accuracy: 50.65 %  || Validation Loss: 0.6994
-------------------------------------------------------------------------


-------------------------------------------------------------------------

After 4 epoch:
Validation Accuracy: 50.26 %  || Validation Loss: 0.7064
-------------------------------------------------------------------------


-------------------------------------------------------------------------

After 5 epoch:
Validation Accuracy: 50.39 %  || Validation Loss: 0.6988
-------------------------------------------------------------------------


-------------------------------------------------------------------------

After 6 epoch:
Validation Accuracy: 49.35 %  || Validation Loss: 0.6977
-------------------------------------------------------------------------


-------------------------------------------------------------------------

After 7 epoch:
Validation Accuracy: 49.09 %  || Validation Loss: 0.6957
-------------------------------------------------------------------------


-------------------------------------------------------------------------

After 8 epoch:
Validation Accuracy: 49.87 %  || Validation Loss: 0.7091
-------------------------------------------------------------------------


-------------------------------------------------------------------------

After 9 epoch:
Validation Accuracy: 50.13 %  || Validation Loss: 0.6979
-------------------------------------------------------------------------


Validation Accuracy: 50.52 %  || Validation Loss: 0.6985
Epoch [10/100], Step [500/50], Loss: 0.1873 | time elapsed: 0.13h |
-------------------------------------------------------------------------

After 10 epoch:
Validation Accuracy: 50.26 %  || Validation Loss: 0.6988
-------------------------------------------------------------------------


-------------------------------------------------------------------------

After 11 epoch:
Validation Accuracy: 53.26 %  || Validation Loss: 0.6944

** Best Performing Model: 53.26 ** 
Saving weights at ./results/roberta/demo/ep_11_stp_551_acc_53.2552_roberta_large.pth
-------------------------------------------------------------------------


-------------------------------------------------------------------------

After 12 epoch:
Validation Accuracy: 52.08 %  || Validation Loss: 0.6927
-------------------------------------------------------------------------


-------------------------------------------------------------------------

After 13 epoch:
Validation Accuracy: 50.65 %  || Validation Loss: 0.6966
-------------------------------------------------------------------------


-------------------------------------------------------------------------

After 14 epoch:
Validation Accuracy: 52.60 %  || Validation Loss: 0.6945
-------------------------------------------------------------------------


-------------------------------------------------------------------------

After 15 epoch:
Validation Accuracy: 50.13 %  || Validation Loss: 0.7136
-------------------------------------------------------------------------


-------------------------------------------------------------------------

After 16 epoch:
Validation Accuracy: 50.91 %  || Validation Loss: 0.6942
-------------------------------------------------------------------------


-------------------------------------------------------------------------

After 17 epoch:
Validation Accuracy: 54.04 %  || Validation Loss: 0.6928

** Best Performing Model: 54.04 ** 
Saving weights at ./results/roberta/demo/ep_17_stp_851_acc_54.0365_roberta_large.pth
-------------------------------------------------------------------------


-------------------------------------------------------------------------

After 18 epoch:
Validation Accuracy: 50.78 %  || Validation Loss: 0.6998
-------------------------------------------------------------------------


-------------------------------------------------------------------------

After 19 epoch:
Validation Accuracy: 52.47 %  || Validation Loss: 0.6962
-------------------------------------------------------------------------


Validation Accuracy: 51.43 %  || Validation Loss: 0.7256
Epoch [20/100], Step [1000/50], Loss: 0.1731 | time elapsed: 0.26h |
-------------------------------------------------------------------------

After 20 epoch:
Validation Accuracy: 51.69 %  || Validation Loss: 0.7254
-------------------------------------------------------------------------


-------------------------------------------------------------------------

After 21 epoch:
Validation Accuracy: 53.78 %  || Validation Loss: 0.6928
-------------------------------------------------------------------------


-------------------------------------------------------------------------

After 22 epoch:
Validation Accuracy: 53.65 %  || Validation Loss: 0.7059
-------------------------------------------------------------------------


-------------------------------------------------------------------------

After 23 epoch:
Validation Accuracy: 53.12 %  || Validation Loss: 0.7082
-------------------------------------------------------------------------


-------------------------------------------------------------------------

After 24 epoch:
Validation Accuracy: 55.73 %  || Validation Loss: 0.7106

** Best Performing Model: 55.73 ** 
Saving weights at ./results/roberta/demo/ep_24_stp_1.2k_acc_55.7292_roberta_large.pth
-------------------------------------------------------------------------


-------------------------------------------------------------------------

After 25 epoch:
Validation Accuracy: 55.60 %  || Validation Loss: 0.7381
-------------------------------------------------------------------------


-------------------------------------------------------------------------

After 26 epoch:
Validation Accuracy: 56.25 %  || Validation Loss: 0.8782

** Best Performing Model: 56.25 ** 
Saving weights at ./results/roberta/demo/ep_26_stp_1.3k_acc_56.2500_roberta_large.pth
-------------------------------------------------------------------------


-------------------------------------------------------------------------

After 27 epoch:
Validation Accuracy: 57.29 %  || Validation Loss: 0.7304

** Best Performing Model: 57.29 ** 
Saving weights at ./results/roberta/demo/ep_27_stp_1.4k_acc_57.2917_roberta_large.pth
-------------------------------------------------------------------------


-------------------------------------------------------------------------

After 28 epoch:
Validation Accuracy: 57.16 %  || Validation Loss: 0.8609
-------------------------------------------------------------------------


-------------------------------------------------------------------------

After 29 epoch:
Validation Accuracy: 59.77 %  || Validation Loss: 0.8471

** Best Performing Model: 59.77 ** 
Saving weights at ./results/roberta/demo/ep_29_stp_1.5k_acc_59.7656_roberta_large.pth
-------------------------------------------------------------------------


Validation Accuracy: 58.46 %  || Validation Loss: 1.0798
Epoch [30/100], Step [1500/50], Loss: 0.0452 | time elapsed: 0.40h |
-------------------------------------------------------------------------

After 30 epoch:
Validation Accuracy: 58.59 %  || Validation Loss: 1.0788
-------------------------------------------------------------------------


-------------------------------------------------------------------------

After 31 epoch:
Validation Accuracy: 60.81 %  || Validation Loss: 1.0477

** Best Performing Model: 60.81 ** 
Saving weights at ./results/roberta/demo/ep_31_stp_1.6k_acc_60.8073_roberta_large.pth
-------------------------------------------------------------------------


-------------------------------------------------------------------------

After 32 epoch:
Validation Accuracy: 58.33 %  || Validation Loss: 1.1676
-------------------------------------------------------------------------


-------------------------------------------------------------------------

After 33 epoch:
Validation Accuracy: 62.50 %  || Validation Loss: 1.1432

** Best Performing Model: 62.50 ** 
Saving weights at ./results/roberta/demo/ep_33_stp_1.7k_acc_62.5000_roberta_large.pth
-------------------------------------------------------------------------


-------------------------------------------------------------------------

After 34 epoch:
Validation Accuracy: 59.90 %  || Validation Loss: 1.1988
-------------------------------------------------------------------------


-------------------------------------------------------------------------

After 35 epoch:
Validation Accuracy: 59.38 %  || Validation Loss: 1.4339
-------------------------------------------------------------------------


-------------------------------------------------------------------------

After 36 epoch:
Validation Accuracy: 60.42 %  || Validation Loss: 1.3058
-------------------------------------------------------------------------


-------------------------------------------------------------------------

After 37 epoch:
Validation Accuracy: 59.11 %  || Validation Loss: 1.3819
-------------------------------------------------------------------------


-------------------------------------------------------------------------

After 38 epoch:
Validation Accuracy: 61.20 %  || Validation Loss: 1.4310
-------------------------------------------------------------------------


-------------------------------------------------------------------------

After 39 epoch:
Validation Accuracy: 61.20 %  || Validation Loss: 1.4858
-------------------------------------------------------------------------


Validation Accuracy: 61.33 %  || Validation Loss: 1.5535
Epoch [40/100], Step [2000/50], Loss: 0.0126 | time elapsed: 0.53h |
-------------------------------------------------------------------------

After 40 epoch:
Validation Accuracy: 61.07 %  || Validation Loss: 1.5602
-------------------------------------------------------------------------

```

**Comment by jaehunjung1** [comment]
Hi, thanks for the quick reply! Turned out that I haven't modified seq_len, and now it seems like valid accuracy is close to the paper. Thanks!

**Comment by Axe--** [comment]
Hey @sharkmir1, could you please share the exact changes (**args**) that helped with convergence for RoBERTa-large.

Since model performance shouldn't be affected by longer sequences (padding), it will be great if we can resolve this error. 

(I will update the Readme accordingly)
